cf-agent/verify_users_pam.c

```diff
@@ -991,7 +991,16 @@ static bool GetGroupInfo (const char *user, const User *u, StringSet **groups_to
 static bool EqualGid(const char *key, const struct group *entry)
 {
     assert(entry != NULL);
-    return (atoi(key) == entry->gr_gid);
+
+    errno = 0;
+    gid_t gid = (gid_t) strtoul(key, (char **) NULL, 10);
+    if (errno != 0)
+    {
+        Log(LOG_LEVEL_ERR, "Failed to read group id from key: %s", GetErrorStr());
+        return false;
```

Comment:
Should this return false on error or should it exit on error?

Reply:
When would this happen? Generally speaking we should not exit on errors. The exception is programming errors / asserts in debug builds, and critical errors (with a critical log message) which prevent CFEngine from functioning.